src/software/thunderscope/thunderscope_main.py

```diff
@@ -331,31 +332,45 @@ def __async_sim_ticker(tick_rate_ms):
             :param tick_rate_ms: The tick rate of the simulation
 
             """
-            world_state = tbots_protobuf.create_world_state(
-                blue_robot_locations=[
-                    cpp_bindings.Point(-3, y) for y in numpy.linspace(-2, 2, NUM_ROBOTS)
-                ],
-                yellow_robot_locations=[
-                    cpp_bindings.Point(3, y) for y in numpy.linspace(-2, 2, NUM_ROBOTS)
-                ],
-                ball_location=cpp_bindings.Point(0, 0),
-                ball_velocity=cpp_bindings.Vector(0, 0),
-            )
-            tscope.simulator_proto_unix_io.send_proto(WorldState, world_state)
+            print("Entered async_sim_ticker function", flush=True)
+            sim_start_buffer = ThreadSafeBuffer(1, SimulationStartedTrigger, False)
+            tscope.simulator_proto_unix_io.register_observer(SimulationStartedTrigger, sim_start_buffer)
 
+            while True:
+                print("Entered sim start loop", flush=True)
+                if not sim_start_buffer.get(block=False):
+                    world_state = tbots_protobuf.create_world_state(
+                        blue_robot_locations=[
+                            cpp_bindings.Point(-3, y) for y in numpy.linspace(-2, 2, NUM_ROBOTS)
+                        ],
+                        yellow_robot_locations=[
+                            cpp_bindings.Point(3, y) for y in numpy.linspace(-2, 2, NUM_ROBOTS)
+                        ],
+                        ball_location=cpp_bindings.Point(0, 0),
+                        ball_velocity=cpp_bindings.Vector(0, 0),
+                    )
+                    tscope.simulator_proto_unix_io.send_proto(WorldState, world_state)
+                    print(f'Python: {world_state=}', flush=True)
+                else:
+                    break
+                
+                time.sleep(10)
```

Comment:
To simplify this, you could use the blocking feature of the `ThreadSafeBuffer` class
```suggestion
            # Block thread until the simulation has started up
            sim_start_buffer.get(block=False) 
            
            world_state = tbots_protobuf.create_world_state(
                blue_robot_locations=[
                    cpp_bindings.Point(-3, y) for y in numpy.linspace(-2, 2, NUM_ROBOTS)
                ],
                yellow_robot_locations=[
                    cpp_bindings.Point(3, y) for y in numpy.linspace(-2, 2, NUM_ROBOTS)
                ],
                ball_location=cpp_bindings.Point(0, 0),
                ball_velocity=cpp_bindings.Vector(0, 0),
            )
            tscope.simulator_proto_unix_io.send_proto(WorldState, world_state)
```